types/evidence.go

```
@@ -768,6 +764,15 @@ type PhantomValidatorEvidence struct {
 
 var _ Evidence = &PhantomValidatorEvidence{}
 
+// NewPhantomValidatorEvidence creates a new instance of the respective evidence 
+func NewPhantomValidatorEvidence(vote *Vote, lastHeightValidatorWasInSet int64) *PhantomValidatorEvidence {
+	if vote == nil || lastHeightValidatorWasInSet <= 0 { return nil }
```

why are u duplicating ValidateBasic logic here?

---

I was just replicating what DuplicateVoteEvidence does - which is add null checks

---

it hides errors. that's not a good style. if anything, the implementation should panic or return an error if someone calls a constructor w/ nil vote

---

I'd remove the check

---

okay. I also thought about having the option to return an error or even to validate basic when we create the evidence

---

but I will remove the checks on all of them except the constructors for DuplicateVoteEvidence and PotentialAmnesiaEvidence which need to order the votes correctly